src/main/java/me/xmrvizzy/skyblocker/utils/UpdateChecker.java

```diff
@@ -0,0 +1,53 @@
+package me.xmrvizzy.skyblocker.utils;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import me.xmrvizzy.skyblocker.SkyblockerMod;
+import net.fabricmc.loader.api.FabricLoader;
+import org.spongepowered.asm.util.VersionNumber;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.net.URLConnection;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class UpdateChecker {
+    public static boolean shouldUpdate(){
+        Pattern pattern = Pattern.compile("v(\\d+)\\.(\\d+)\\.(\\d+)");
+        Pattern localPattern = Pattern.compile("(\\d+)\\.(\\d+)\\.(\\d+)");
+        Matcher matcher;
+        VersionNumber localVersion = null;
+        VersionNumber latestVersion = null;
+        Boolean shouldUpdate = false;
+
+        try{
+            URL url = new URL("https://api.modrinth.com/v2/project/skyblocker-liap/version");
+            URLConnection request = url.openConnection();
```

Comment:
URLConnection blocks on read. This could cause the game to freeze if the site you are trying to reach is down or have slow internet.

Reply:
Should be fixed